hamilton/include/hamilton/hybrid_planner.h

```diff
@@ -0,0 +1,133 @@
+    double time_offset; // = 0.0; Used inside toROSJointTrajectory()
+  };
+
+  /*  =============================== TrajectorySegment Struct ===============================
+  * Struct to store the unplanned segments, and categorize them as free and process path. 
+  * Note: The third type "FREE_MOTION_START_END_ONLY" isn't used right now.
+  * The idea is to call the MoveIt planner via a MotionPlanning Request in this case, via the PlanHybrid method.  
+  * TODO Add more data structures in TrajectorySegment - joint vals, descartes type itself, etc */
+
+  struct TrajectorySegment
+  {
+    TrajectorySegmentType type_;      
+    std::vector<geometry_msgs::Pose> waypoints_geom_msgs_;
+  }; 
+
+  /*  =============================== Hybrid Planner Class ===============================
+   * Provides a group of functions for planning a robot path using Moveit and  the Descartes Planning Library */
+
+  class HybridPlanner
+  {
+    public:
+      HybridPlanner();
+      virtual ~HybridPlanner();
+      // Initialize (descartes) robot model and planners
+      void init();
+
+    protected:
+      // Append unplanned free space/process path segments defined by "waypoints' to unplanned_trajectory_ as a TrajectorySegment object. 
+      // TODO extend to other types - joints, descartes types, etc.
+    	void appendFreeMotionTrajectorySegment(std::vector<geometry_msgs::Pose>& waypoints);
+    	void appendProcessPathTrajectorySegment(std::vector<geometry_msgs::Pose>& waypoints);
+
+      // Appends unplanned free space segment, defined by start and end point only as a TrajectorySegment object. 
+      void appendFreeMotionTrajectorySegment(geometry_msgs::Pose start_pose, geometry_msgs::Pose end_pose);
+
+      // Iterates through the unplanned_trajectory_vector, and calls MoveIt! and Descartes planners as required.
+      // Saves the planned, joint space trajectory in overall_robot_traj_, which can be send for execution to the controller.
+      // TODO Not Implemented: Makes a motion planning request if the TrajetorySegemnt.type_ == FREE_MOTION_START_END_ONLY
+    	void PlanHybrid();
+
+      //====== Helper functions =====//
+      /* Create an axial trajectory pt from a given tf transform */
+      descartes_core::TrajectoryPtPtr tfToAxialTrajectoryPt(const tf::Transform& nominal, double discretization);  
+
+      // Generates an completely defined (zero-tolerance) cartesian point from a pose
+      descartes_core::TrajectoryPtPtr makeCartesianPoint(const Eigen::Affine3d& pose);
+
+      // Generates a cartesian point with free rotation about the Z axis of the EEF frame
+      descartes_core::TrajectoryPtPtr makeTolerancedCartesianPoint(const Eigen::Affine3d& pose);
+
+      // Translates a descartes trajectory to a ROS joint trajectory
+      trajectory_msgs::JointTrajectory toROSJointTrajectory(const TrajectoryVec& trajectory, const std::vector<std::string>& joint_names, double time_delay);
```

Comment:
Yes, they can. For `toROSJointTrajectory`, I'll use `config_.joint_names` instead of passing `const std::vector<std::string>& joint_names`